did_webvh/core/resolver.py

```diff
@@ -250,44 +324,38 @@ async def resolve_state(
                     aborted_err = e
                     next_state = None
 
-                if found:
-                    # no extra verification needed
-                    continue
-
                 if not state:
-                    if not next_state:
-                        if aborted_err:
-                            raise aborted_err
-                        raise ValueError("Empty document history")
-                    if version_time and next_state.timestamp > version_time:
-                        raise ValueError(f"Cannot resolve versionTime: {version_time}")
+                    # we initially loop twice so that the next line is available (if any)
                     continue
 
-                if version_id:
-                    if state.version_id == version_id:
-                        if version_time and state.timestamp > version_time:
-                            raise ValueError(
-                                "Specified `versionId` not valid at specified"
-                                " `versionTime`"
-                            )
+                if not found:
+                    if version_number:
+                        if state.version_number == version_number:
+                            if version_time and state.timestamp > version_time:
+                                # "Specified `versionId` not valid at `versionTime`"
+                                return None
+                            if version_id is not None and state.version_id != version_id:
+                                # "`versionId` mismatch with specified `versionNumber`"
+                                return None
+                            found = state
+                    elif version_time and (
+                        (next_state and next_state.timestamp > version_time)
+                        or (not next_state and not aborted_err)
+                    ):
+                        if state.timestamp > version_time:
+                            # "Resolved version not valid at specified `versionTime`"
```

Comment:
it looks like the behavior here changed from raising an error to returning a None value. What's the reason for this change?

Reply:
This is to indicate a `notFound` error where before an exception raised here would return `invalidDid`. It will need further updates to return specific `problemDetails` in the 1.0 update (raising a specific exception subclass).